python_modules/libraries/dagster-dbt/dagster_dbt_tests/components/test_dbt_project_component.py

```diff
@@ -305,17 +310,32 @@ def test_state_path(
     comp = load_component_for_test(
         DbtProjectComponent,
         {
-            "dbt": {
-                "project_dir": dbt_path,
+            "project": {
+                "project_dir": str(dbt_path),
                 "state_path": "state",
                 "profile": "profile",
                 "target": "target",
             },
         },
     )
-    state_path = comp.dbt.state_path
+    state_path = comp.cli_resource.state_path
     assert state_path
     assert Path(state_path).relative_to(dbt_path.resolve())
-    assert comp.project.state_path == Path(state_path)
+    assert comp.project.state_path
+    assert comp.project.state_path.resolve() == Path(state_path)
     assert comp.project.target == "target"
     assert comp.project.profile == "profile"
+
+
+def test_python_interface(dbt_path: Path):
+    context = ComponentLoadContext.for_test()
+    assert DbtProjectComponent(
+        project=DbtProject(dbt_path),
+    ).build_defs(context)
+
+    defs = DbtProjectComponent(
+        project=DbtProject(dbt_path),
+        translation=lambda spec, _: spec.replace_attributes(tags={"python": "rules"}),
+    ).build_defs(context)
+    assets_def = defs.get_assets_def(AssetKey("stg_customers"))
+    assert assets_def.get_asset_spec(AssetKey("stg_customers")).tags["python"] == "rules"
```

Comment:
minimal pythonic interface example here, anything specific you think would be useful to demonstrate ?

Reply:
yea, the bag of stuff from the dbt manifest

Reply:
I think for translation, having a one-shot process makes the most sense, but we still have the "transforms" (asset post processors) layer in the DefsFolderComponent for that sort of thing. 

Basically, I want the "translation" process to be philosophically simple (we have some data in an external system, we go through it in a loop and you can tell us how you want to represent that in dagster).

Then we can have a more general "you have some information in dagster, do you want to modify it in any way?" process, but the relevant APIs there will be fairly different.